### PR TITLE
Improve error message if flatmap function doesn't return a strategy

### PR DESCRIPTION
**Pull request by kx-chen**
This improves the error message if the `flatmap` function doesn't return a `SeachStrategy`

### CODE REVIEW

File: hypothesis-python/tests/cover/test_searchstrategy.py [hunk 1]
```diff
@@ -93,3 +95,9 @@ def test_can_map_nameless():
 def test_can_flatmap_nameless():
     f = nameless_const(just(3))
     assert repr(f) in repr(integers().flatmap(f))
+
+
+def test_flatmap_with_invalid_expand():
```
Comment (kx-chen) [inline hypothesis-python/tests/cover/test_searchstrategy.py]: A bit unsure if this is the proper way to do things, please advise if not.

Comment (Zalathar) [reply hypothesis-python/tests/cover/test_searchstrategy.py]: I think replacing the `do_draw` with a simple `.example()` should do the trick here.

File: hypothesis-python/src/hypothesis/searchstrategy/flatmapped.py [hunk 2]
```diff
@@ -40,7 +41,10 @@ def __repr__(self):
 
     def do_draw(self, data):
         source = data.draw(self.flatmapped_strategy)
-        return data.draw(self.expand(source))
+        expanded_source = self.expand(source)
+
```
Comment (Zac-HD) [inline hypothesis-python/src/hypothesis/searchstrategy/flatmapped.py]: Tiny style nitpick, but I'd drop this blank line.

File: hypothesis-python/RELEASE.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+RELEASE_TYPE: patch
+
+This release ensures that functions passed to SearchStrategy.flatmap return a SeachStrategy
```
Comment (Zac-HD) [inline hypothesis-python/RELEASE.rst]: The existence of `SearchStrategy` is only public to help type hints, so I'd rephrase this a bit - and say thanks!  It's a bit awkward to thank yourself, so suggestion below, along with docs hyperlink syntax.

```rst
RELEASE_TYPE: patch

This patch improves the error message if the function ``f`` in 
:ref:`s.flatmap(f) <flatmap>` does not return a strategy.

Thanks to Kai Chen for this change!
```

**Comment by Zac-HD** [comment]
Released as [Hypothesis 4.14.2](https://hypothesis.readthedocs.io/en/latest/changes.html#v4-14-2)  :shipit::tada:
